src/Plugin/testsite_builder/EntityType/Media.php

```diff
@@ -51,6 +51,21 @@ protected function getBundleConfig($bundle_id, array $bundle_config) {
     return $config;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function postCreateBundle(string $bundle_id, array $bundle_config, array $created_fields): void {
+    /** @var \Drupal\media\MediaTypeInterface $type */
+    $type = $this->entityTypeManager->getStorage('media_type')->load($bundle_id);
+
+    $config = $type->getSource()->getConfiguration();
+    $key = $bundle_config['source']['source_field_index'];
```

Comment:
This option could be empty, for example `generic` type, so it would be good to check if `source_field_index` exists first. Also if it does not exist, we should have some fallback.